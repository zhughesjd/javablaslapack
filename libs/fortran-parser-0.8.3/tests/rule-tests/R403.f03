! Test intrinsic_type_spec (R403)
integer :: a
INTEGER(4) :: b
real :: c
real(4) :: d
DOUBLE precision :: e
doublePRECISION :: f
complex :: g
COMPLEX(4) :: h
double complex :: i
doublecomplex :: j
CHARACTER :: k
character(1) :: l
logical :: m
LOGICAL(1) :: n
integer foo


end

