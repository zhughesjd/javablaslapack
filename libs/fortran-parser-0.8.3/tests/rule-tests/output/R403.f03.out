********************************************
args[1]: R403.f03
R1101:main-program__begin:
R 501:type-declaration-stmt__begin:
R 403:intrinsic-type-spec: type=400 hasKindSelector=false
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@6,45:45='a',<206>,2:11]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 406:int-literal-constant: digitString=[@10,55:55='4',<10>,3:8] kindParam=null
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=400 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@15,61:61='b',<206>,3:14]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 403:intrinsic-type-spec: type=401 hasKindSelector=false
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@21,71:71='c',<206>,4:8]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 406:int-literal-constant: digitString=[@25,78:78='4',<10>,5:5] kindParam=null
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=401 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@30,84:84='d',<206>,5:11]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 403:intrinsic-type-spec: type=402 hasKindSelector=false
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@38,106:106='e',<206>,6:20]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 403:intrinsic-type-spec: type=402 hasKindSelector=false
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@44,127:127='f',<206>,7:19]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 403:intrinsic-type-spec: type=404 hasKindSelector=false
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@50,140:140='g',<206>,8:11]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 406:int-literal-constant: digitString=[@54,150:150='4',<10>,9:8] kindParam=null
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=404 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@59,156:156='h',<206>,9:14]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 403:intrinsic-type-spec: type=403 hasKindSelector=false
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@67,176:176='i',<206>,10:18]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 403:intrinsic-type-spec: type=403 hasKindSelector=false
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@73,195:195='j',<206>,11:17]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 403:intrinsic-type-spec: type=405 hasKindSelector=false
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@79,210:210='k',<206>,12:13]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 406:int-literal-constant: digitString=[@83,222:222='1',<10>,13:10] kindParam=null
R 402:type-param-value: hasExpr=true hasAsterisk=false hasColon=false
R 424:char-selector: kindOrLen1=302 kindOrLen2=300 hasAsterisk=false
R 403:intrinsic-type-spec: type=405 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@88,228:228='l',<206>,13:16]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 403:intrinsic-type-spec: type=406 hasKindSelector=false
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@94,241:241='m',<206>,14:11]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 406:int-literal-constant: digitString=[@98,251:251='1',<10>,15:8] kindParam=null
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=406 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@103,257:257='n',<206>,15:14]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 403:intrinsic-type-spec: type=400 hasKindSelector=false
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@107,267:269='foo',<206>,16:8]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=15
R9999:end-of-stmt:
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=false hasInternalSubprogramPart=false
Parser exiting normally
********************************************
